tests/typ/compiler/field.typ

```diff
@@ -148,3 +148,18 @@
 #test((top + start).y, top)
 #test((bottom + end).y, bottom)
 #test((horizon + center).y, horizon)
+
+---
+#{
+  let object = none
+  // Error: 3-9 cannot mutate fields on none
```

Comment:
is this really the clearest error message? isn't it even one step more. not just that you cannot mutate fields, none doesn't even _have_ fields.

Reply:
Yeah, I agree the error message could be more descriptive. Initially, in this PR, I had only distinguished between types with static fields and types without static fields, as it's simple to do that (a simple call to `fields::fields_on`). I can probably make this more fine-grained, however, using a proper `match` in the error message generation.

Will update the code later. Thanks for the feedback!

Reply:
Ok, should be better now :+1: